sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -134,10 +137,14 @@ final class Decimal extends Ordered[Decimal] with Serializable {
       // result. For example, the precision of 0.01 equals to 1 based on the definition, but
       // the scale is 2. The expected precision should be 3.
       this._precision = decimal.scale + 1
+      this._scale = decimal.scale
+    } else if (decimal.scale < 0 && !SQLConf.get.allowNegativeScaleOfDecimalEnabled) {
+      this._precision = decimal.precision - decimal.scale
+      this._scale = 0
```

Comment:
Do you mean allowing negative scale here?

Reply:
Yeah. But the negative scale is not from end-user, but from java BigDecimal(e.g. `1E10` will has precision=2 and scale=-9 using java BigDecimal).

Reply:
Oh I see. Thanks.